src/core/meshcop/dataset_manager_ftd.cpp

```diff
@@ -333,26 +320,8 @@ otError ActiveDataset::CreateNewNetwork(otOperationalDataset &aDataset)
         preferredChannels = supportedChannels;
     }
 
-    aDataset.mChannel     = preferredChannels.ChooseRandomChannel();
-    aDataset.mChannelMask = supportedChannels.GetMask();
-
-    aDataset.mPanId = Mac::GenerateRandomPanId();
-
-    snprintf(aDataset.mNetworkName.m8, sizeof(aDataset.mNetworkName), "OpenThread-%04x", aDataset.mPanId);
-
-    aDataset.mComponents.mIsActiveTimestampPresent = true;
-    aDataset.mComponents.mIsMasterKeyPresent       = true;
-    aDataset.mComponents.mIsNetworkNamePresent     = true;
-    aDataset.mComponents.mIsExtendedPanIdPresent   = true;
-    aDataset.mComponents.mIsMeshLocalPrefixPresent = true;
-    aDataset.mComponents.mIsPanIdPresent           = true;
-    aDataset.mComponents.mIsChannelPresent         = true;
-    aDataset.mComponents.mIsPskcPresent            = true;
-    aDataset.mComponents.mIsSecurityPolicyPresent  = true;
-    aDataset.mComponents.mIsChannelMaskPresent     = true;
-
-exit:
-    return error;
+    return aDataset.GenerateRandom(preferredChannels.ChooseRandomChannel(), supportedChannels.GetMask(),
```

Comment:
What about passing `preferredChannels` and let `GenerateRandom()` select a random channel?

Reply:
In the new push, I changed it to pass `aInstance` and then have the `Dataset::Info::GenerateRandom()` method itself read the preferred/supported masks and decide on channel.